src/edu/gatech/cs2340/trydent/math/MathTools.java

```diff
@@ -0,0 +1,106 @@
+package edu.gatech.cs2340.trydent.math;
+
+import java.lang.reflect.Array;
+
+/**
+ * Library of advanced math tools, primarily linear algebra. 
+ * <p>
+ * To be used by advanced developers only.
+ * @author Garrett Malmquist
+ */
+public class MathTools {
+	
+	// We don't want people instantiating this class.
+	private MathTools() {}
+	
+	/**
+	 * Performs an (n-1) dimensional Bezier interpolation over n points. (Advanced functionality).
+	 * @param t - interpolation parameter - t=0.0 will return the 1st point, 
+	 * 										t=1.0 will return the last point.
+	 * @param points - One or more points to interpolate between.
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	public static <T extends BaseVector<?>> T bezierInterpolation(double t, T ... points) {
```

Comment:
Come to think of it, I should probably just name this 'bezier', to be consistent with all following methods. That, or have 'interpolation' after all the methods already there (and rename the lerps to 'linear interpolation'). Since these methods are explicitly for advanced stuff (and internally will be useful for animations), I think it's reasonable to be a bit more concise for the sake of readability. I think it's a lot easier to read "hermite(...)" or "lerp(...)" than "hermiteInterpolation(...)" or "linearInterpolation(...)".